crates/sui-snapshot/src/lib.rs

```diff
@@ -292,3 +318,29 @@ pub fn create_file_metadata(
     };
     Ok(file_metadata)
 }
+
+pub async fn setup_db_state(
+    epoch: u64,
+    perpetual_db: Arc<AuthorityPerpetualTables>,
+    checkpoint_store: Arc<CheckpointStore>,
+    committee_store: Arc<CommitteeStore>,
+) -> Result<()> {
+    // This function should be called once state accumulator based hash verification
+    // is complete and live object set state is downloaded to local store
+    let system_state_object = get_sui_system_state(&perpetual_db)?;
+    let new_epoch_start_state = system_state_object.into_epoch_start_state();
+    let next_epoch_committee = new_epoch_start_state.get_sui_committee();
+    let last_checkpoint = checkpoint_store
+        .get_epoch_last_checkpoint(epoch)
+        .expect("Error loading last checkpoint for current epoch")
+        .expect("Could not load last checkpoint for current epoch");
+    let epoch_start_configuration =
+        EpochStartConfiguration::new(new_epoch_start_state, *last_checkpoint.digest());
+    perpetual_db
+        .set_epoch_start_configuration(&epoch_start_configuration)
+        .await?;
+    committee_store.insert_new_committee(&next_epoch_committee)?;
+    checkpoint_store.update_highest_synced_checkpoint(&last_checkpoint)?;
+    checkpoint_store.update_highest_executed_checkpoint(&last_checkpoint)?;
+    Ok(())
```

Comment:
We will also need to update the `root_state_hash_by_epoch` table to insert the last root state hash accumulator. Since orchestration will need to accumulate and confirm the live object set via the snapshot, it should already have that, so we could probably simply pass that in here. Could you update this function to take that as an arg and write to the table?

Reply:
Done, added this bit above:
```
perpetual_db.insert_root_state_hash(epoch, last_checkpoint.sequence_number, accumulator)?;
```